src/main/java/sb/persistence/dao/OrderDAO.java

```diff
@@ -0,0 +1,94 @@
+package sb.persistence.dao;
+
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+import sb.domain.entity.Order;
+import sb.domain.entity.User;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class OrderDAO {
+    
+    private NamedParameterJdbcTemplate namedJdbcTemplate;
+    private BeanPropertyRowMapper<Order> rowMapper;
+
+    public OrderDAO(NamedParameterJdbcTemplate namedJdbcTemplate) {
+        this.namedJdbcTemplate = namedJdbcTemplate;
+        rowMapper = new BeanPropertyRowMapper<>(Order.class);
+    }
+
+    private final String SQL_SELECT_ALL = "select id, status, submitted_by, submitted_at, updated_at " +
+                                            "from orders";
+
+    private final String SQL_SELECT_BY_ID = "select id, status, submitted_by, submitted_at, updated_at " +
+                                            "from orders where id = :id";
+
+    private final String SQL_SELECT_BY_USER = "select id, status, submitted_by, submitted_at, updated_at " +
+                                                "from orders where submitted_by = :submittedBy";
+
+    private final String SQL_INSERT = "insert into orders(status, submitted_by, submitted_at, updated_at) " +
+                                        "values (:status, :submittedBy, :submittedAt, :updatedAt)";
+
+    private final String SQL_DELETE_BY_ID = "delete from orders where id = :id";
+
+    private final String SQL_UPDATE_STATUS_BY_ID = "update orders set status = :status, updated_at = :updatedAt " +
+                                                    "where id = :id";
+
+
+    public List<Order> getAll() {
+        return namedJdbcTemplate.query(SQL_SELECT_ALL, rowMapper);
+    }
+
+    public Optional<Order> get(int id) {
+        MapSqlParameterSource map = new MapSqlParameterSource()
+                .addValue("id", id);
+
+        Order order = null;
+        try {
+            order = namedJdbcTemplate.queryForObject(SQL_SELECT_BY_ID, map,rowMapper);
+        } catch (DataAccessException ignored){}
```

Comment:
1. Generally, you should not ignore exceptions. At least log it
2. If the exception is thrown because of empty result set, please catch this concrete exception and return an empty `Optional` directly from `catch` block.

Reply:
1. fix: log it
2. concretize the exception(EmptyResultDataAccessException). If exception were thrown, method  would return Optional.ofNullable(order)(not in the catch block) where order=null, and later it wold be handled in a more appropriate place. don't understand why better way is return an empty Optional directly from catch block. If the thing is in the readability may add final block with return Optional.ofNullable(order)